lib/tapyrus/script/script.rb

```diff
@@ -518,17 +521,27 @@ def p2sh_addr
     # generate cp2pkh address. if script dose not cp2pkh, return nil.
     def cp2pkh_addr
       return nil unless cp2pkh?
+
+      color_id = chunks[0].pushed_data.bth
+      return nil unless Tapyrus::Color::ColorIdentifier.parse_from_payload(color_id.htb)&.valid?
+
       hash160 = chunks[4].pushed_data.bth
       return nil unless hash160.htb.bytesize == 20
-      Tapyrus.encode_base58_address(hash160, Tapyrus.chain_params.address_version)
+
```

Comment:
How about doing these checks with `cp2pkh?`?
This can avoid the situation where `cp2pkh?` returns true but the address is nil.

Reply:
Fixed at c617971.

And I remove size check of hash160 data in these methods because hash160 should be already checked in cp2pkh? methods (or cp2sh?, p2pkh? p2sh?)